code/game/jobs/job/marine/squads.dm

```diff
@@ -765,3 +765,248 @@
+/datum/squad_overwatch_info
+	var/leader_count = 0
+	var/rto_count = 0
+	var/spec_count = 0
+	var/medic_count = 0
+	var/engi_count = 0
+	var/smart_count = 0
+	var/marine_count = 0
+	var/living_count = 0
+
+	var/almayer_count = 0
+	var/SSD_count = 0
+	var/helmetless_count = 0
+
+
+	//Formatted for the Overwatch Console
+	var/squad_sorted_text_overwatch = ""
+	var/living_sorted_text_overwatch = ""
+
+	//Formatted for the Squad Monitor
+	var/squad_sorted_text_crew_monitor = ""
+
+	//Formatted for the Groundside Operations Computer
+	var/squad_sorted_text_groundside_operations = ""
+
+/**
+ * A proc for printing info for the Ovarwatch Console, the Squad Monitor, and the Groundside Operations computer
+ * 
+ * Arguments:
+ * * z_hidden_overwatch - ignores a Z-level (Overwatch Console)
+ * * marine_filter - ignored marines (Overwatch Console)
+ * * marine_filter_enabled - if the filter is enabled (Overwatch Console)
+ * * dead_hidden - whether to hide dead marines (Overwatch Console)
+ * * user - user of the device, used to determine distance between them and squad members (Squad Monitor)
+ * * z_hidden_groundside_ops - TRUE means we won't display user information if they are not on the ground (Groundside Operations)
+*/
+/datum/squad/proc/get_squad_overwatch_info(var/z_hidden_overwatch = OVERWATCH_HIDE_NONE, var/list/marine_filter = list(), var/marine_filter_enabled = TRUE, var/dead_hidden = FALSE, var/mob/user = null, z_hidden_groundside_ops = FALSE)
+	var/datum/squad_overwatch_info/info = new()
+
+	var/leader_text_overwatch = ""
+	var/rto_text_overwatch = ""
+	var/spec_text_overwatch = ""
+	var/medic_text_overwatch = ""
+	var/engi_text_overwatch = ""
+	var/smart_text_overwatch = ""
+	var/marine_text_overwatch = ""
+	var/misc_text_overwatch = ""
+
+	var/conscious_text_overwatch = ""
+	var/unconscious_text_overwatch = ""
+	var/dead_text_overwatch = ""
+
+	var/leader_text_crew_monitor = ""
+	var/rto_text_crew_monitor = ""
+	var/spec_text_crew_monitor = ""
+	var/medic_text_crew_monitor = ""
+	var/engi_text_crew_monitor = ""
+	var/smart_text_crew_monitor = ""
+	var/marine_text_crew_monitor = ""
+	var/misc_text_crew_monitor = ""
+
+	var/leader_text_groundside_operations = ""
+	var/rto_text_groundside_operations = ""
+	var/spec_text_groundside_operations = ""
+	var/medic_text_groundside_operations = ""
+	var/engi_text_groundside_operations = ""
+	var/smart_text_groundside_operations = ""
+	var/marine_text_groundside_operations = ""
+	var/misc_text_groundside_operations = ""
+
+	var/SL_z //z level of the Squad Leader
+	if(squad_leader)
+		var/turf/SL_turf = get_turf(squad_leader)
+		SL_z = SL_turf.z
+
+	for(var/X in marines_list)
+		if(!X)
+			continue //just to be safe
+		var/mob_name = "unknown"
+		var/mob_state = ""
+		var/role = "unknown"
+		var/act_sl = ""
+		var/fteam = ""
+		var/dist = "<b>???</b>"
+		var/area_name = "<b>???</b>"
+		var/mob/living/carbon/human/H
```

Comment:
human and should also be after the typecheck

Reply:
Some code still references the /mob outside the if statement. Probably should be fixed, but I didn't want to change the logic, only to squish the code.